support/cas-server-support-rest-x509/src/main/java/org/apereo/cas/support/x509/rest/config/X509RestConfiguration.java

```diff
@@ -46,20 +47,36 @@ public RestHttpRequestCredentialFactory x509RestMultipartBody() {
     public RestHttpRequestCredentialFactory x509RestRequestHeader() {
         return new X509RestHttpRequestHeaderCredentialFactory(x509CertificateExtractor.getIfAvailable());
     }
-    
+
+    @Bean
+    public RestHttpRequestCredentialFactory x509RestTlsClientCert() {
+        return new X509RestTlsClientCertCredentialFactory();
+    }
+
     @Override
     public void configureCredentialFactory(final ChainingRestHttpRequestCredentialFactory factory) {
         val restProperties = casProperties.getRest();
         val extractor = x509CertificateExtractor.getIfAvailable();
         val headerAuth = restProperties.isHeaderAuth();
         val bodyAuth = restProperties.isBodyAuth();
-        LOGGER.debug("is certificate extractor available? = {}, headerAuth = {}, bodyAuth = {}",
-            extractor, headerAuth, bodyAuth);
+        val tlsClientAuth = restProperties.isTlsClientAuth();
+        LOGGER.debug("is certificate extractor available? = {}, headerAuth = {}, bodyAuth = {}, tlsClientAuth = {}",
+                     extractor, headerAuth, bodyAuth, tlsClientAuth);
+
+        if (tlsClientAuth && (headerAuth || bodyAuth)) {
+            LOGGER.warn("X509 REST configuration seems inconsistant as activating headerAuth or bodyAuth supposes "
+                        + "that the CAS server is not directly reachable by clients (see security warning in the "
+                        +"documentation) and activating tlsClientAuth suppose the opposite.");
```

Comment:
- What security warning in the documentation? Please remove this bit.
- `...and activating tlsClientAuth assumes the opposite`.

Reply:
In docs/cas-server-documentation/protocol/REST-Protocol.md there is a "usage warning" about allowing connections to the CAS server only from trusted hosts. At this moment, the warning applies to the "bodyAuth" method but in my opinion, it should also apply to the "headerAuth" method, as both methods allow obtaining TGTs without proof of private key ownership.
On the contrary, tlsClientAuth method needs a direct connection from CAS client to CAS server (for tls hand-check to occur, which forces to provide a proof of private key ownership) . The idea is to warn about the incompatibility of the first two methods with the last one.

Reply:
You should either remove that statement from the logs or you need to repeat the warning to some degree in the logs again, and explain the inconsistency as you do above in your comment. It's unclear which warning exactly that is, and 2 months from now when the page is removed, updated or has 4 other warnings this statement in the code becomes that much more confusing and is impossible to track or maintain.